.github/workflows/style.yml

```diff
@@ -0,0 +1,24 @@
+# Run style checks
+
+name: style
+
+on:
+  - pull_request
+  - workflow_dispatch
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          cache: "pip"
+          cache-dependency-path: ".github/workflows/qc-requirements.txt"
```

Comment:
From the documentation at https://github.com/actions/setup-python/blob/main/README.md, it looks as though they recommend always providing a Python version. Please could you add that here? I think `python-version: '3.12' ` would work.

Reply:
Also, let's drop the `cache-dependency-path` line:

```suggestion
```